rapids-4-spark-tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ProfileMain.scala

```diff
@@ -150,4 +147,61 @@ object ProfileMain extends Logging {
 
     0
   }
+
+  /**
+   * Function to evaluate the event logs to be processed.
+   *
+   * @param eventDir       directory containing the event logs
+   * @param filterNLogs    number of event logs to be selected
+   * @param matchlogs      keyword to match file names in the directory
+   * @param eventLogsPaths Array of event log paths
+   * @return event logs to be processed
+   */
+  def processAllPaths(
+      filterNLogs: ScallopOption[String],
+      matchlogs: ScallopOption[String],
+      eventLogsPaths: List[String]): Seq[Path] = {
+
+    val allPaths: ArrayBuffer[Path] = ArrayBuffer[Path]()
+    val allPathsWithTimestamp: Map[Path, Long] = Map.empty[Path, Long]
+
+    for (pathString <- eventLogsPaths) {
+      val (paths, pathsWithTimestamp) = ProfileUtils.stringToPath(pathString)
+      if (paths.nonEmpty) {
+        allPaths ++= paths
+        allPathsWithTimestamp ++= pathsWithTimestamp
+      }
+    }
+
+    // Filter the eventlogs to be processed based on the criteria. If it is not provided in the
+    // command line, then return all the event logs processed above.
+    if (matchlogs.isDefined || filterNLogs.isDefined) {
+      var sortedResult: LinkedHashMap[Path, Long] = LinkedHashMap.empty[Path, Long]
+      if (filterNLogs.isDefined) {
+        val numberofEventLogs = filterNLogs.toOption.get.split("-")(0).toInt
+        val criteria = filterNLogs.toOption.get.split("-")(1)
+        if (criteria.equals("newest")) {
+          sortedResult = LinkedHashMap(allPathsWithTimestamp.toSeq.sortWith(_._2 > _._2): _*)
+        } else if (criteria.equals("oldest")) {
+          sortedResult = LinkedHashMap(allPathsWithTimestamp.toSeq.sortWith(_._2 < _._2): _*)
+        } else {
+          logError("Criteria should be either newest or oldest")
+          System.exit(1)
+        }
+
+        if (matchlogs.isDefined) {
```

Comment:
I think you can do the filter first and then sort so you sort less. also seems like those could be separate (ie not under the same if)... first do match if its specified and then filterNLogs if specified.

Reply:
Done